tsdb/track.json

```diff
@@ -32,92 +32,17 @@
       {%- if corpus is defined and corpus == "split16" %}
         "documents": [
           {
-            "source-file": "documents-split-0.json.bz2",
-            "document-count": 7289607,
-            "uncompressed-bytes": 8252385790
-          },
-          {
-            "source-file": "documents-split-1.json.bz2",
-            "document-count": 7289607,
-            "uncompressed-bytes": 8252715218
-          },
-          {
-            "source-file": "documents-split-2.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252482193
-          },
-          {
-            "source-file": "documents-split-3.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8253066083
-          },
-          {
-            "source-file": "documents-split-4.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252686254
-          },
-          {
-            "source-file": "documents-split-5.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252842205
-          },
-          {
-            "source-file": "documents-split-6.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252839714
-          },
-          {
-            "source-file": "documents-split-7.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8253327643
-          },
-          {
-            "source-file": "documents-split-8.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252994659
-          },
-          {
-            "source-file": "documents-split-9.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8253163552
-          },
-          {
-            "source-file": "documents-split-10.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252932917
-          },
-          {
-            "source-file": "documents-split-11.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252752704
-          },
-          {
-            "source-file": "documents-split-12.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8253251862
-          },
-          {
-            "source-file": "documents-split-13.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8253166730
-          },
-          {
-            "source-file": "documents-split-14.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252927958
-          },
-          {
-            "source-file": "documents-split-15.json.bz2",
-            "document-count": 7289606,
-            "uncompressed-bytes": 8252803345
+            "source-file": "documents-split16.json.bz2",
```

Comment:
Question: how does Rally know how to split this file into 16 parts? (and each part having the correct date range)

Reply:
Splitting the file in 16 is the default behavior of Rally, provided we configure it to use 16 bulk indexing clients. I rearranged the original file to make sure that each of the 16 splits was starting from beginning to end. Given the previous split16 work, all I had to do was:

> cat documents-split-0.json documents-split-1.json documents-split-2.json documents-split-3.json documents-split-4.json documents-split-5.json documents-split-6.json documents-split-7.json documents-split-8.json documents-split-9.json documents-split-10.json documents-split-11.json documents-split-12.json documents-split-13.json documents-split-14.json documents-split-15.json > documents-split16.json

Does that make sense?

Reply:
I see, and each of the `documents-split-N.json` files has an equal size, so that Rally knows exactly how to split the `documents-split16.json` file?

Reply:
Yes, each bulk indexing client will split and mmap the file independently based on its id (from 0 to 15) and the number of lines (which is why we have to provide document-count).

Reply:
👍